File: packages/expo-updates/android/src/main/java/expo/modules/updates/manifest/Manifest.java
package expo.modules.updates.manifest;

import expo.modules.updates.db.entity.AssetEntity;
import expo.modules.updates.db.entity.UpdateEntity;

import org.json.JSONObject;

import java.util.ArrayList;

public interface Manifest {
  UpdateEntity getUpdateEntity();
  ArrayList<AssetEntity> getAssetEntityList();
  JSONObject getRawManifestJson();
}

File: packages/expo-updates/android/src/main/java/expo/modules/updates/manifest/Manifest.kt
package expo.modules.updates.manifest

import expo.modules.updates.db.entity.AssetEntity
import expo.modules.updates.db.entity.UpdateEntity
import expo.modules.updates.manifest.raw.RawManifest
import org.json.JSONObject

interface Manifest {
  val updateEntity: UpdateEntity?
  val assetEntityList: List<AssetEntity?>?
  val rawManifest: RawManifest
  val serverDefinedHeaders: JSONObject?
  val manifestFilters: JSONObject?
  val isDevelopmentMode: Boolean
}
